### PR TITLE
Add new benchmark to track cost of dependencies to a User

### PR DESCRIPTION
**Pull request by jonahwilliams**
While the dependencies_count benchmark is great for tracking the overall health of the repo, it doesn't specifically measure the costs of dependencies on our users (nor does the technical debt benchmark). This is important because while our dependencies have a cost to developers on the framework, they can have an even larger cost to our users.

Specific Examples:
- A new user is introduced to Flutter and looks up a tutorial on the internet. They follow the directions exactly which includes adding a new dependency on `package:http` to the pubspec. The version of Flutter they use has a dependency that is incompatible with it. Frustrated and without enough knowledge of the pub ecosystem to resolve it, they move on.

- A Dart power user has written a new code generation package and wants to make sure as many developers can use it as possible. The package must depend on `package:analyzer` and they know that Flutter has a pinned version.  So they spend extra time each week making sure it is compatible, instead of responding to issues or addressing new feature requests.

This is also important to track separately from the overall dependency count. A certain refactor which drastically reduces the dependencies of users (cough cough) may increase the overall dependency count of the repo, by splitting a single package up.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
cc @tvolkert - we talked about measuring this during the OKR meeting

**Review by tvolkert — COMMENTED** [review]
👍 to the benchmark, but maybe it'd be better suited in the `analyzer_benchmark`?  wrt consumer deps, it's not really technical debt.

But I could see arguments both ways.  Thoughts?

**Comment by jonahwilliams** [comment]
I don't have a strong opinion, happy to move it there

**Comment by jonahwilliams** [comment]
Though purely from an implementation perspective, it is easier to leave where it is currently :)

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -107,6 +115,25 @@ class UpdatePackagesCommand extends FlutterCommand {
     final bool isPrintPaths = argResults['paths'];
     final bool isPrintTransitiveClosure = argResults['transitive-closure'];
     final bool isVerifyOnly = argResults['verify-only'];
+    final bool isConsumerOnly = argResults['consumer-only'];
+
+    // "consumer" packages are those our users are likely to rely on, and represents
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/update_packages.dart]: "those our users are likely to rely on" sounds wishy washy - how about "those that come from packages that constitute our public API (e.g. flutter, flutter_test, flutter_driver, flutter_localizations)"

**Review by tvolkert — APPROVED** [review]

**Pull request merged**